src/index.ts

```diff
@@ -156,7 +184,9 @@ class CommandBuilder {
       this.initCommandInstance(this.parser, initedCommand)
     }
 
-    this.context = this.parser.parse(argv)
+    await this.autocomplete(options)
+
+    this.context = await this.parser.parse(test ? argv : argv.slice(2))
```

Comment:
Please change the `argv` input to unified (slice the `process.argv` before this) and delete the `test` boolean.